crates/spirv-std/src/textures.rs

```diff
@@ -46,6 +46,65 @@ impl Image2d {
             result
         }
     }
+    #[spirv_std_macros::gpu_only]
+    #[cfg(feature = "const-generics")]
+    /// Sample the image at a coordinate by a lod
+    pub fn sample_by_lod<V: Vector<f32, 4>>(
+        &self,
+        sampler: Sampler,
+        coordinate: impl Vector<f32, 2>,
+        lod: f32,
+    ) -> V {
+        let mut result = Default::default();
+        unsafe {
+            asm!(
+                "%image = OpLoad _ {this}",
+                "%sampler = OpLoad _ {sampler}",
+                "%coordinate = OpLoad _ {coordinate}",
+                "%lod = OpLoad _ {lod}",
+                "%sampledImage = OpSampledImage _ %image %sampler",
+                "%result = OpImageSampleExplicitLod _ %sampledImage %coordinate Lod %lod",
+                "OpStore {result} %result",
+                result = in(reg) &mut result,
+                this = in(reg) self,
+                sampler = in(reg) &sampler,
+                coordinate = in(reg) &coordinate,
+                lod = in(reg) &lod
```

Comment:
`lod` technically doesn't have to go through a pointer, but, shrug, whatever, consistency is neat too

Reply:
I honestly wasn't sure, so I did it out of ignorance moreso than consistency, but I'll take it